### PR TITLE
Add convenience function date

### LINKED ISSUE #4581: Add convenience function date
This function `date(x)` should be the same as `CAST(x AS date)`.

### PR DESCRIPTION
**Pull request by prateek1306**
Fixes #4581 
Here `date(x)` is not being redirected to `cast(x as date)`. Rather it has a separate flow. I am not sure whether that is correct?
Queries work fine though. Have also written test cases.
```
presto> select date('2015-09-10');
   _col0
------------
 2015-09-10
(1 row)
```

```
presto> select date(timestamp '2012-08-08 01:00:13');
   _col0
------------
 2012-08-08
(1 row)
```

```
presto> select date(1);
Query 20160222_124319_00005_rhwby failed: CAST to date cannot be applied to bigint
```

Please review.

### Edit(after review changes):

```
presto> select date(TIMESTAMP '2001-08-22 03:29:05.321 +03:30');
   _col0
------------
 2001-08-22
```

```
presto> select date(TIMESTAMP '2001-08-22 03:29:05.321');
   _col0
------------
 2001-08-22
```

```
presto> select date('2001-08-22');
   _col0
------------
 2001-08-22
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/DateFunction.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Primitives;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Map;
+
+import static com.facebook.presto.metadata.Signature.typeParameter;
+import static java.lang.invoke.MethodType.methodType;
+
+public class DateFunction
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/DateFunction.java]: No need to implement it with a full class. This mechanism is for when additional flexibility is needed (e.g., for parametric functions).

Take a look at `com.facebook.presto.operator.scalar.DateTimeFunctions`

**Comment by prateek1306** [comment]
@martint I have done the changes. Also added test cases.
Expected input: `date(varchar)` , `date(timestamp)` , `date(timestamp with time zone)`

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java [hunk 2]
```diff
@@ -204,6 +205,33 @@ public static long fromISO8601Timestamp(ConnectorSession session, @SqlType(Stand
         return packDateTimeWithZone(parseDateTimeHelper(formatter, iso8601DateTime.toStringUtf8()));
     }
 
+    @ScalarFunction("date")
+    @SqlType(StandardTypes.DATE)
+    public static long dateFromVarchar(ConnectorSession session, @SqlType(StandardTypes.VARCHAR) Slice date)
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java]: It turns out you can achieve the same result by annotating the function that implements the cast from varchar->date with `@ScalarFunction("date")` (I checked to be sure). Take a look at `DateOperators.castFromSlice`, `TimestampOperators.castToDate` and `TimestampWithTimezoneOperators.castToDate`.

**Comment by prateek1306** [comment]
Thanks @martint 
Changes done.

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**